airqo-platform/AirQo-api
# Updating Device Registry

- What does this PR do?
> 1. Storage of newly created device on local/production DB
> 2. Addition of description field in request data, removal of  ISP, manufacturer, phone number from the description field stored
> 3. Retrieving all devices from the database
> 4. Pending work: An endpoint for deploying a device

- What JIRA task is related to this PR?
The link to the JIRA board task is [here](https://airqoteam.atlassian.net/browse/NET-14?atlOrigin=eyJpIjoiYjgxZmZmNjljODBiNDU3OWFmMWI5MDU4M2Y1ZDZhODUiLCJwIjoiaiJ9).
- How do I test out this PR?
Git clone this repo `cd AirQo-api/src/device-registry` Then run `npm install` to get all the necessary dependencies Afterwards run `npm run dev` or `npm run stage` to test out the end points using any REST client tool available to you.
- Which end points should are ready for testing?
> base url: http://localhost:3000/
> [**GET**] [**Get All Devices**] **partial url:** _/api/v1/devices?limit="number"&skip="number"_ 
> [**POST**]  [**create Thing**] **partial url:** _api/v1/devices/ts_

API documentation is [here ](https://docs.google.com/spreadsheets/d/1-NGTDBtTcIi8Fg2UYSWjcK6geKrZdopz5kxRFsCEMPs/edit#gid=897606332)for more details. Please visit the **device-registry** sheet.
